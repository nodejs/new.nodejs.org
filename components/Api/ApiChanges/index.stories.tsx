import ApiChanges from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof ApiChanges>;
type Meta = MetaObj<typeof ApiChanges>;

const version = 'v10.0.0';
export const Default: Story = {
  args: {
    update: {
      type: 'added',
      version: [version],
    },
    changes: [
      {
        version: version,
        'pr-url': 'https://github.com/nodejs/node/pulls',
        description: 'test',
      },
    ],
  },
};

export default { component: ApiChanges } as Meta;
