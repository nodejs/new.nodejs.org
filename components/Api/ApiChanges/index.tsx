import { FormattedMessage } from 'react-intl';
import styles from './index.module.scss';
import { parseApiDocsVersion } from '../../../util/parseApiDocsVersion';
import type { ApiChange, ApiUpdate } from '../../../types';
import type { FC } from 'react';

type ChangesProps = {
  update: ApiUpdate;
  changes: ApiChange[];
};

const Changes: FC<ChangesProps> = ({ update, changes }) => (
  <details className={styles.changesComponent}>
    <summary>
      <strong>
        <FormattedMessage id="docs.api.history" />
      </strong>
    </summary>
    <table>
      <thead>
        <tr>
          <th>
            <FormattedMessage id="docs.api.history.version" />
          </th>
          <th>
            <FormattedMessage id="docs.api.history.changes" />
          </th>
        </tr>
      </thead>
      <tbody>
        {changes.map(({ version, description }) => (
          <tr key={`${version.toString()}-${description}`}>
            <td>{parseApiDocsVersion(version)}</td>
            <td>{description}</td>
          </tr>
        ))}
        <tr>
          <td>{parseApiDocsVersion(update.version)}</td>
          <td>
            <FormattedMessage
              id="docs.api.addedIn"
              values={{ version: parseApiDocsVersion(update.version) }}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </details>
);

export default Changes;
