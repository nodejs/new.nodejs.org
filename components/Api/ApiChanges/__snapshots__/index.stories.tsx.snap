// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api/ApiChanges Default smoke-test 1`] = `
<details class="ApiChanges_changesComponent__8zdMP">
  <summary>
    <strong>
      History
    </strong>
  </summary>
  <table>
    <thead>
      <tr>
        <th>
          Version
        </th>
        <th>
          Changes
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          v10.0.0
        </td>
        <td>
          test
        </td>
      </tr>
      <tr>
        <td>
          v10.0.0
        </td>
        <td>
          Added in: v10.0.0
        </td>
      </tr>
    </tbody>
  </table>
</details>
`;
